azurerm/internal/services/bot/bot_connection_resource.go

```diff
@@ -112,9 +112,37 @@ func resourceArmBotConnectionCreate(d *schema.ResourceData, meta interface{}) er
 		}
 	}
 
+	serviceProviderName := d.Get("service_provider_name").(string)
+	serviceProviderFound := false
+	var availableProviders []string
+
+	serviceProviders, err := client.ListServiceProviders(ctx)
+	if err != nil {
+		return fmt.Errorf("listing Bot Connection service provider: %+v", err)
+	}
+
+	if serviceProviders.Value == nil {
+		return fmt.Errorf("no available service provider %+v", serviceProviders)
```

Comment:
this isn't an actionable error for users, so I guess we can make this:

```suggestion
		return fmt.Errorf("no service providers were returned from the Azure API")
```